db-services/db-diagram-service/src/main/java/com/qreal/wmp/db/diagram/server/DiagramDbServiceHandler.java

```diff
@@ -98,6 +131,63 @@ public TFolder getFolderTree(String username) throws TNotFound {
         } catch (NotFoundException e) {
             throw new TNotFound(username, "FolderTree for specified user not found.");
         }
-        return folder.toTFolder();
+        return folder.toTFolder(username);
+    }
+
+    @Override
+    public void shareFolderTo(String username, TFolder tFolderToShare) throws TException {
```

Comment:
shouldn't we use some kind of IDs for addressing users instead of just user names? but I guess that's a task for future pull-requests